### PR TITLE
Add mosquitto 1.6.10

### PR DESCRIPTION
**Pull request by uilianries**
Add Eclipse Mosquitto (very useful for embedded and IoT)

Specify library name and version:  **mosquitto/1.6.10**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
@uilianries please fix the build and then re-request the review(s)

**Comment by prince-chrismc** [comment]
@uilianries @SSE4 I would love to see this added, I worked on it a little bit https://github.com/prince-chrismc/conan-center-index/tree/mosquitto I got it working on MSVC 2019

Please consider my two commits! (I can not make a pr against bincrafters repo)

**Comment by Croydon** [comment]
@prince-chrismc You can: https://github.com/bincrafters/conan-center-index/compare/mosquitto...prince-chrismc:mosquitto

**Comment by prince-chrismc** [comment]
Thanks! I dont remember this at all 🏎️  

Please consider bincrafters/conan-center-index#19

**Comment by prince-chrismc** [comment]
Oh no! This really is stale

File: recipes/mosquitto/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch(self):
+        if not self.options.shared and not self.options.get_safe("fPIC"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                  "POSITION_INDEPENDENT_CODE 1",
+                                  "POSITION_INDEPENDENT_CODE 0")
+        if self.settings.os == "Windows":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
```
Comment (a4z) [inline recipes/mosquitto/all/conanfile.py:64]: this explains (and fixes) my problem in https://github.com/conan-io/conan-center-index/pull/3949
however, I think a patch should be created, and the change should be suggested upstream to the mosquitto project ,  hopefully I will find the time for that

**Pull request closed without merging**